### PR TITLE
Profiling data

### PR DESCRIPTION
**Pull request by spapinistarkware**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

No profiling data

## What is the new behavior?

Profiling data for each test in pprof format when running tests with the `--trace-run` flag.

## Other information

Generating profiling data is **slow** and consequently disabled by default! When you want to use the tool, just run the target test with `pytest <target test> --trace-run`

### CODE REVIEW

**Review by abdelhamidbakhta — APPROVED** [review]
Lgtm

**Review by abdelhamidbakhta — APPROVED** [review]
Lgtm

File: tests/utils/utils.py [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+            program_base=1,
+            debug_info=runner.get_relocated_debug_info(),
+        )
+
+        profile = profile_from_tracer_data(tracer_data)
+        _profile_datas[traceit._context] = profile
+    return runner, syscall_handler
+
+
+ExecuteEntryPoint._run = override_run
```
Comment (ClementWalter) [inline tests/utils/utils.py:30]: can you move this after the import statements? and don't forget to run `make format` to make sure the ordering is the right one

File: tests/utils/utils.py [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+from starkware.cairo.lang.tracer.tracer_data import TracerData
+from starkware.starknet.business_logic.execution.execute_entry_point import (
+    ExecuteEntryPoint,
+)
+
+# Override _run() entrypoint for hooking and profiling cairo runner.
+old_run = ExecuteEntryPoint._run
+
+
+def override_run(*args, **kwargs):
```
Comment (ClementWalter) [inline tests/utils/utils.py:11]: could be better to make it a decorator, like : 

```
def trace_run(run):
    def override_run(*args, **kwargs):
    # use the run param here and not `old_run`
          return

    return override_run

```

and in `conftest.py` we use a fixture with `autouse=True` to enable the behavior during the test suite

```
@pytest.fixutre(autouse=True)
def overide_run():
    ExecuteEntryPoint._run = trace_run(ExecuteEntryPoint._run)
    return
```

**Review by Eikix — APPROVED** [review]
lgtm

**Pull request merged**